From b3122d11685c57517913cf224139613b4e78a56d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 04 Aug 2021 21:18:37 +0300
Subject: [PATCH] [SW-51761] gaudi: security: allow host sm access

We have moved to another solution for protecting the SM from unsupported
host accesses.
Remove the PCIe RRs protection to allow the LKD direct access to the
SM address space.

Change-Id: I599b426d8ce1adb23c5bafecef2d37c690a411fd
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/security.c b/arch/gaudi/security.c
index 99cdb48..06fa7bf 100644
--- a/arch/gaudi/security.c
+++ b/arch/gaudi/security.c
@@ -1395,13 +1395,9 @@
 	LBW_RR_DBG_ETR,
 };
 
-#define SYNC_MNGR_DIST		0x20000
-#define SYNC_MNGR_SIZE		0x10000
-
 static void preboot_secure_lbw(void)
 {
-	int i;
-	u64 min, max, base;
+	u64 min, max;
 
 	/*
 	 * NOTE: The LBW and PCIe RRs HIT is: min < hit < max
@@ -1495,14 +1491,6 @@
 	 */
 	max = mmTPC7_EML_CS_BASE + TPC7_EML_CS_MAX_OFFSET - 1;
 	preboot_secure_lbw_rr_if(LBW_RR_DBG_ETR, min, max);
-
-	/* SYNC_MNGR: PCIe only! PCIe RR_28 up to PCIe RR_31 */
-	for (i = 0; i < 4; i++) {
-		base = mmSYNC_MNGR_GLBL_W_S_BASE + SYNC_MNGR_DIST * i;
-		min = base - 1;
-		max = base + SYNC_MNGR_SIZE;
-		preboot_secure_pcie_rr(28 + i, min, max);
-	}
 }
 
 static void preboot_hbw_rr_set_rtr_wr(int rr, int rtr, u64 base, u64 mask,
@@ -1675,7 +1663,7 @@
 	 * PCIe IF HBW RRs are for direct access from PCIe IF to PSOC domain:
 	 * STM, SPI Flash, Scratchpad SRAM, PCIe FW SRAM.
 	 * SPSRAM space for fit is used in dynamic RR.
-	 * Assign the PCIe IF RR_16, RR_22, RR_24 (0-15 and 28-31 are allocated
+	 * Assign the PCIe IF RR_16, RR_22, RR_24 (0-15 are allocated
 	 * for LBW) for this job.
 	 * NOTE: The PCIe RRs use min/max concept and HIT is: min < hit < max
 	 */
